snap/snapcraft.yaml

```diff
@@ -626,6 +627,10 @@ parts:
           install -DT "./cmd/$service/$service" "$SNAPCRAFT_PART_INSTALL/bin/$service"
 
           case "$service" in
+          "core-metadata")
+              install -DT "./cmd/core-metadata/res/configuration.toml" "$SNAPCRAFT_PART_INSTALL/config/core-metadata/res/configuration.toml"
+              install -DT "./cmd/core-metadata/res/uom.toml" "$SNAPCRAFT_PART_INSTALL/config/core-metadata/res/uom.toml"
+          ;;
```

Comment:
This is good as it copies also `uom.toml` to the snap package (readonly, [`SNAP`](https://snapcraft.io/docs/environment-variables) env var). But we also need to add code to copy it over to the snap's writable area ([`SNAP_DATA`](https://snapcraft.io/docs/environment-variables) env var) on install time. The given path `UOM_UOMFILE: $SNAP_DATA/config/core-metadata/res/uom.toml` expects it to be under the writable area.

Currently only `configuration.toml` files get copied:
https://github.com/edgexfoundry/edgex-go/blob/e3ca38a5709b6fd134cd3d7a24fe413a337b80c3/snap/local/hooks/cmd/install/install.go#L118-L143

This is really an unnecessary complication in how the snap is packaged. Maybe we should refactor and simply copy the whole `res` directory.

For now, a quick fix would be to update the referenced code to copy `uom.toml` from `SNAP` to `SNAP_DATA` for core-metadata. I'll work on it and commit to the same PR (since the permission has been granted by the author during PR submission.)

Reply:
sorry I wasn't aware you're working on it.
I've added a quick fix: https://github.com/edgexfoundry/edgex-go/compare/4c4951218dd81a254464f99d357ca8617bed2473..3aaec565c9b77e4c31d71bd8720703868403a2fd

Reply:
Thanks for the fix. I was wondering why I couldn't push. I was testing locally so it took some time.

You force-pushed so it isn't clear what exactly you changed. But the install.go part looks good.